universum/analyzers/uncrustify.py

```diff
@@ -1,200 +1,202 @@
+    previous_match = matching_blocks[0]
+    for match in matching_blocks[1:]:
+        block = _get_mismatching_block(previous_match, match, src, target)
+        previous_match = match
+        if not block:
+            continue
+        line, before, after = block
+        path: Path = src_file.relative_to(Path.cwd())
+        message = _get_issue_message(before, after)
+        result.append(utils.ReportData(
+            symbol="Uncrustify Code Style issue",
+            message=message,
+            path=str(path),
+            line=line
+        ))
+
+    return result
+
+
+def _get_issue_message(before: str, after: str) -> str:
+    # The maximum number of lines to write separate comments for
+    # If exceeded, summarized comment will be provided instead
+    max_lines = 11
+    diff_size = len(before.splitlines())
+    if diff_size > max_lines:
+        message = f"\nLarge block of code ({diff_size} lines) has issues\n"
+    else:
+        # Message with before&after
+        message = f"\nOriginal code:\n```diff\n{before}```\n" + \
+                  f"Uncrustify generated code:\n```diff\n{after}```\n"
+    return message
 
-        files = self.parse_files()
-        try:
-            cmd = sh.Command("uncrustify")
-            cmd("-c", self.settings.cfg_file, "--prefix", self.settings.output_directory, files)
-        except sh.ErrorReturnCode as e:
-            sys.stderr.write(str(e) + '\n')
 
-        issues_loads = []
-        for file_name in files:
-            issues_loads.extend(self.get_file_issues(file_name))
-        utils.output_report_to_file(issues_loads, self.settings.result_file)
-        return 1 if issues_loads else 0
+def _get_mismatching_block(previous_match: difflib.Match,  # src[a:a+size] = target[b:b+size]
+                           match: difflib.Match,
+                           src: List[str], target: List[str]
+                           ) -> Optional[Tuple[int, str, str]]:
+    previous_match_end_in_src = previous_match.a + previous_match.size
+    previous_match_end_in_target = previous_match.b + previous_match.size
+    match_start_in_src = match.a
+    match_start_in_target = match.b
+    if previous_match_end_in_src == match_start_in_src:
+        return None
+    line = match_start_in_src
+    before = _get_text_for_block(previous_match_end_in_src - 1, match_start_in_src, src)
+    after = _get_text_for_block(previous_match_end_in_target - 1, match_start_in_target, target)
+    return line, before, after
 
 
-def form_arguments_for_documentation():
-    return UncrustifyAnalyzer.define_arguments()
+def _get_text_for_block(start: int, end: int, lines: List[str]) -> str:
+    return _replace_invisible_symbols(''.join(lines[start: end]))
 
 
-def main():
-    analyzer_namespace = UncrustifyAnalyzer.define_arguments().parse_args()
-    analyze = UncrustifyAnalyzer(analyzer_namespace)
-    return analyze.execute()
+def _replace_invisible_symbols(line: str) -> str:
```

Comment:
this function used to have a bug, where line was returned immediately after first replace